test/resources/qemu-guest-agent.py

```diff
@@ -0,0 +1,302 @@
+    boolean return value will be True.
+
+    Note: DOES NOT wrap the Process.terminate_process function. This is because the Process version only kills
+    processes on the host, not the guest.
+
+    Examples:
+    ${stdout}    ${stderr}    ${rc}=    Terminate Process    vm_name    pid
+    Should Be Equal As Integers    ${rc}    -15
+    Terminate Process    vm_name    pid    kill=True
+    """
+
+    return _do_kill(vm_name, pid, -15 if not kill else -9)
+
+
+def get_process_result(vm_name: str, pid: str | int) -> (dict, bool):
+    """
+    :param vm_name:        The name of the VM to execute the command on
+    :type vm_name:         str
+    :param pid:            The process ID returned by qemu-agent-command.execute.guest-exec
+    :type pid:             str | int
+    :return:               The stdout, stderr, and exit code of the guest process for the given PID
+    :rtype:                dict
+    :return:               True if the process has exited, False otherwise
+    :rtype:                bool
+    :raises RuntimeError:  If the PID is not specified
+
+    ``get_process_result`` gets the results of the qemu-agent-command for a given process ID, which is returned by the
+    qemu-agent-command.execute.guest-exec call immediately after execution. The PID is used to retrieve that process's
+    stdout, stderr, and exit code. If the process is still running, the exitcode, stdout, and stderr will be None and
+    the boolean return value will be False. If the process has exited, the exit code will be returned and the boolean
+    return value will be True.
+
+    Note: DOES NOT wrap the Process.get_process_result function. This is because the qemu-agent-command does not
+    return the exit code of the command executed on the guest. Instead, it returns the exit code of the qemu-agent '
+    itself (which is always 0). This is a known limitation of the qemu-agent.
+    """
+    if not pid:
+        raise RuntimeError('PID is not specified')
+
+    agent_cmd_wrapper = {
+        'execute': 'guest-exec-status',
+        'arguments': {
+            'pid': pid
+        }
+    }
+    virsh_args = (
+        f'--connect=qemu:///system qemu-agent-command --domain={vm_name}'
+        f'--cmd={json.dumps(agent_cmd_wrapper)}'
+    )
+
+    # The status of the qemu-guest-command is returned as a json struct following this format:
+    #     {
+    #         "return": {
+    #             "out-data": "stdout",
+    #             "err-data": "stderr",
+    #             "exitcode": 0
+    #             "exited": true
+    #         }
+    #     }
+    result: ExecutionResult = Process().run_process('virsh', virsh_args)
+    # extract the qemu-agent-command return value from the virsh stdout
+    guest_result = json.loads(result.stdout)['return']
+    # Rename keys to match the Process library's return values for consistency. Return 'exited': bool separately to
+    # avoid confusion with the Process library's return.
+    return {
```

Comment:
This could return a DotDict to make it easier to access the contents from RF keywords. There's an example in the DataFormats.py library.

Reply:
keywords now return DotDicts where applicable